packages/core/src/transports/initialize.ts

```diff
@@ -86,24 +88,43 @@ export function initializeTransports(
     });
   };
 
-  const execute: Transports['execute'] = (item) => {
-    if (!paused) {
-      let actualItem = item;
+  const send = (item: TransportItem | TransportItem[]) => {
+    let items = isArray(item) ? item : [item];
+    for (const hook of beforeSendHooks) {
+      // TODO: Make BeforeSendHook work with item array
+      const modified = items.map((bshItem) => hook(bshItem)).filter((i) => i !== null) as TransportItem[];
 
```

Comment:
This introduces a breaking change for users of the `beforeSend` hook.

The [beforeSend hook currently expects a TransportItem instead of a list of items](https://github.com/grafana/faro-web-sdk/blob/eebcee0adc9e2e6f18fc2d1c836c10df15e052eb/packages/core/src/transports/types.ts#L11).

`export type BeforeSendHook<P = APIEvent> = (item: TransportItem<P>) => TransportItem<P> | null;`

Reply:
Don't forget to align beforeSend to handle lists.
We also need to [upate the cloud docs](https://grafana.com/docs/grafana-cloud/frontend-observability/faro-web-sdk/faq/#does-the-faro-sdk-have-a-sampling-capability-to-rate-limit-the-telemetry-data-sent-to-grafana-cloud).